### PR TITLE
feat: password reset flow

### PR DESCRIPTION
**Pull request by datchley**
- added /auth/password/reset for requesting a password reset given an email and reset link/url
- added /auth/password/reset/change that can be called by the form at the previous reset link with a reset token, email, password & confirm to actually reset password.

### CODE REVIEW

File: src/api/controllers/auth.controller.js [hunk 1]
```diff
@@ -68,6 +82,34 @@ exports.passwordReset = async (req, res, next) => {
...
+    const { email, password, confirm, token } = req.body;
+    // Ensure user is valid
+    const user = await User.getByEmail(email);
+    // Find a matching reset token and clear out tokens
+    const valid = await User.matchAndClearAllTokens({ email, token });
+    // Update password for user
+    user.password = password;
+    user.save();
+
+    const userTransformed = user.transform();
```
Comment (ceane) [inline src/api/controllers/auth.controller.js:102]: what does `user.transform` do ?

File: src/api/validations/auth.validation.js [hunk 2]
```diff
@@ -33,12 +33,35 @@ module.exports = {
...
         .required(),
       url: Joi.string()
-        .uri()
+        // .uri()
         .trim()
         .required(),
     },
   },
 
+  passwordResetChange: {
```
Comment (ceane) [inline src/api/validations/auth.validation.js:42]: Looks good that we have password now, this means we'll have to update this on the client end too, correct?

**Review by ceane — APPROVED** [review]
I have a few non-blocking comments. This is good to go.

File: src/api/controllers/auth.controller.js (hunk 1 above)
Comment (datchley) [reply src/api/controllers/auth.controller.js:102]: `transform` is in the User model and part of the original boilerplate I started from. It takes a user object, and basically filters out any system specific fields and cleans it up for returning to the client in reponses.

File: src/api/validations/auth.validation.js (hunk 2 above)
Comment (datchley) [reply src/api/validations/auth.validation.js:42]: Yes, so moving the client apps to support this basic api-boilerplate is definitely something to move toward (at least for projects that would use this boilerplate).

**Pull request merged**